libmproxy/proxy/root_context.py

```diff
@@ -47,6 +74,7 @@ def next_layer(self, top_layer):
         layer = self._next_layer(top_layer)
         return self.channel.ask("next_layer", layer)
 
+    @tls_sni_check_ignore
```

Comment:
> I was doing it because there were a few exit points where a TlsLayer was being created (e.g., for scripts) and I wasn't sure if they might be upgraded with SNI (and subsequently ignored if desired). 

Not sure if I understand your point. How does the decorator fix this?

> If I move the `check_tcp` up will that impact any of the earlier cases? before it was happening at step 4, after CONNECT etc.

Sorry, not sure what I thought when writing my previous comment :smile:.
Please keep the order as-is. We can start with checking for `is_tls_record_magic`, then parse ClientHello if check_ignore is not None, then continue with check_ignore and the rest as-is.